sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -601,6 +601,264 @@ manager, with `__aenter__`, `__aexit__`, and `close` methods.
+
+To customize parts of the polling method, you can create a subclass which uses `LROBasePolling` and overrides necessary methods.
+If significant customization is necessary, use `azure.core.polling.PollingMethod` and implement all the necessary methods.
+
+Example: Create an LRO method which will poll for when a file gets uploaded successfully (greatly simplified)
+
+
+```python
+class CustomLROPolling(PollingMethod):
+    def __init__(self, polling_interval=30, **kwargs) -> None:
+        self._polling_interval = polling_interval
+        self._resource = None
+        self._finished = False
+        self._status = "InProgress"
+        self._kwargs = kwargs
+
+    def initialize(self, client, initial_response, deserialization_callback):
+        """Set the initial status of this LRO.
+        """ 
+        self.client = client
+        self.file_id = response.file_id
+        self._pipeline_response = self._initial_response = initial_response
+        self._deserialization_callback = deserialization_callback
+        self._command = functools.partial(self.client.get_upload_file, file_id=self.file_id, **self._kwargs)
+
+    def status(self) -> str:
+        """Return the current status as a string.
+
+        :rtype: str
+        """
+        return "Succeeded" if self._finished else "InProgress"
+
+    def finished(self) -> bool:
+        """Is this polling finished?
+
+        :rtype: bool
+        """
+        return self._finished
+
+    def resource(self) -> JSON:
+        """Return the built resource."""
+        return self._resource
+
+    def run(self) -> None:
+        """The polling loop"""
+        while not self.finished():
+            try:
+                self._resource = self._command()
+                self._finished = True
+            except ResourceNotFoundError:
+                pass
+            if not self.finished():
+                time.sleep(self._polling_interval)
+
+    def get_continuation_token(self) -> str:
+        import pickle
+
+        return base64.b64encode(pickle.dumps(self._initial_response)).decode("ascii")
+
+    @classmethod
+    def from_continuation_token(cls, continuation_token: str, **kwargs) -> Tuple[Any, Any, Callable]:
+        try:
+            client = kwargs["client"]
+        except KeyError:
+            raise ValueError("Need kwarg 'client' to be recreated from continuation_token")
+
+        import pickle
+
+        initial_response = pickle.loads(base64.b64decode(continuation_token))  # nosec
+        # Restore the transport in the context
+        initial_response.context.transport = client._client._pipeline._transport  # pylint: disable=protected-access
+        return client, initial_response, lambda: None
+```
+
+And now, to plug into the client code:
+
+```python
+class ServiceOperations:
+
+    def begin_upload(self, data: AnyStr, **kwargs) -> LROPoller[JSON]:
+        continuation_token = kwargs.pop("continuation_token", None)
```

Comment:
Does this mean that a `continuation_token` kwarg is the official rehydration policy for LROs?
I know quite a few of our APIs do this today - just not sure we ever formalized it as a guideline.

Reply:
Discussed offline. TBD, but leaving it in the doc for now.